src/js/weather/helpers.js

```diff
@@ -1,42 +1,50 @@
 const urlWithParams = (url, params) => {
-    let urlFetch = new URL(url);
-    Object.keys(params).forEach(key => urlFetch.searchParams.append(key, params[key]));
-    return urlFetch;
+  let urlFetch = new URL(url);
+  Object.keys(params).forEach(key =>
+    urlFetch.searchParams.append(key, params[key])
+  );
+  return urlFetch;
+};
+
+const importAllFiles = r => {
+  let images = {};
+  r.keys().map(
+    item => (images[item.replace('./', '').replace('.png', '')] = r(item))
```

Comment:
yyyy
ale o co tutaj chodzi?

Reply:
Używana jest tylko raz - do importowanie plików z ikonami dla pogody (weatherView.js).
`const images = importAllFiles(
  require.context('../../images/weather-icon/', false)
);`
Gdy uruchomimy aplikacje to nie mamy obrazków, bo trzeba je zaimportować.
Ale informację jaką chce konkretnie ikonę mam po pobraniu danych z API. A moja informacja wyglądała tak: '50n'.
Po zaimportowaniu obrazków mam za to ciekawą nazwę pliku: 'cafa1ee1c50faccf5ea980cd083b07cf.png'.
A ja musiałam to wstawić do _src_ tagu _img_.
A ta magiczna metoda zwraca mi obiekt, gdzie:
- klucz to nazwa po której łapie, gdy przyjdą informacje z API - usunięcie elementu ścieżki './' oraz rozszerzenie pliku '.png'.
- a wartość to nazwa pliku.
`50d: "cafa1ee1c50faccf5ea980cd083b07cf.png"
50n: "cafa1ee1c50faccf5ea980cd083b07cf.png"`
Plików z ikonami jest 18. Miałam jeszcze pomysł z switch'em i importowania wszystkiego osobno.
Zagrożenie: niezgodna nazwa obrazu ikony z zwracaną wartością z API.

Reply:
ale ja nie o tym :eyes:
.map() zwraca tablice, mysle ze warto zrobic tak, zeby zwracala od razu tablice ktora jest Ci potrzebna :eyes: zamiast pushowac do zupelnie innego arraya warto tutaj zwrocic tablice idealna (o ile sie da oczywiscie)

Reply:
Zmieniłam na .forEach(), tworząc i tak nowy obiekt. 
Można zrobić idealnie i to działa, ale nie jestem przekonana co do usuwania elementów z niej (chyba, że można zmienić nazwę klucza bezpośrednio):
`const importAllFiles = r => {
  r.keys().map(item => {
    r[item.replace('./', '').replace('.png', '')] = r(item);
    delete r(item);
  });
  return r;
};`
Inny sposób to nie robić tutaj replace (z usuwaniem), tylko dodawać to do przychodzącej wartości z API.